src/app/clusters/resource-monitoring-server/resource-monitoring-cluster-objects.h

```diff
@@ -141,37 +143,72 @@ struct TypeInfo
 };
 } // namespace LastChangedTime
 
-namespace ReplacementProductStruct {
-enum class Fields : uint8_t
+// A struct used during reads of the ReplacementProductList to store a single list instance we request
+// from the application.
+//
+// Inherit from an auto-generated struct to pick up the implementation bits, but make
+// it private inheritance so people can't accidentally use this struct where the other
+// is expected.
+struct GenericType : private HepaFilterMonitoring::Structs::ReplacementProductStruct::Type
```

Comment:
`Attributes::GenericType` is a very odd place to put this.

How about putting it under `Clusters::ResourceMonitoring` and calling it `ReplacementProductStruct`?  That would have the virtue of matching the spec.